### ISSUE TITLE
Avoid unnecessary file creation when waiting on file change.

### ISSUE THREAD

**Issue by aloiscochard** [sbt/sbt#2185]
For some reason, I had to monitor file creation/deletion in my `/tmp` directory while using `sbt`, and I realize something weird.

When I do `~test:compile`, while waiting, `sbt` keep creating and deleting a same file over and over (two times per second).

Why is that? couldn't that be avoided?

Thanks

![screenshot - 08282015 - 01 52 11 pm](https://cloud.githubusercontent.com/assets/215476/9545693/7ca97efe-4d8c-11e5-9075-8cf16b15b06b.png)

**Comment by eed3si9n** [comment]
Thanks for this report.

**Comment by fury0928** [comment]
Hi @eed3si9n may I work on this issue as well?

**Comment by eed3si9n** [comment]
@eureka928 Yes please. This issue is from 10+ years ago, so it may or may not still happen, but we'd be happy to accept PR or investigation.

**Comment by fury0928** [comment]
I investigated this issue against the current sbt 2.x codebase.

### Findings

The original report was from sbt 0.13.x (2015), which used `PollingWatchService` that polled the filesystem every 500ms — matching the reported 2-times-per-second temp file creation pattern.

In sbt 2.x, the watch mechanism has been completely rewritten:

1. **NIO-based file watching by default** — sbt 2.x uses `FileSystems.getDefault.newWatchService()` on Linux/Windows (kernel-level inotify/kqueue) and `MacOSXWatchService` (swoval/closewatch) on macOS (`Watched.scala:58-68`). These use OS-level notifications with no temp file I/O.

2. **No temp files in the watch hot path** — I searched every `createTempFile`/`createTempDirectory` call in the codebase. None are in the continuous watch polling loop. The idle loop in `Watch.apply` just blocks on `monitor.poll(duration)`, a blocking NIO call with no file I/O.

3. **`PollingWatchService` is opt-in only** — Only used when `sbt.watch.mode=polling` is explicitly set. The default is NIO on Linux/Windows and closewatch on macOS.

### Conclusion

This issue appears to be resolved in sbt 2.x as a side effect of the watch mechanism rewrite from polling-based to NIO/swoval-based file watching. The repeated temp file creation/deletion in `/tmp` should no longer occur with the default watch service.

**Comment by eed3si9n** [comment]
Thanks for the detective work!
